streams/src/main/java/org/apache/kafka/streams/query/StateQueryResult.java

```diff
@@ -97,11 +97,15 @@ public void addResult(final int partition, final QueryResult<R> r) {
      * prior observations.
      */
     public Position getPosition() {
-        Position position = Position.emptyPosition();
-        for (final QueryResult<R> r : partitionResults.values()) {
-            position = position.merge(r.getPosition());
+        if (globalResult != null) {
```

Comment:
Not sure I understand this: in this PR we've effectively turned off the only caller of "result.setGlobalResult(r);" so this should never hit any more right?

Reply:
That's correct, but since the old version of this method was incorrect, I figured I'd go ahead and fix it.

Reply:
Okay, but still just to clarify, at least for now we do not expect the `if` block to ever trigger right?

Reply:
That's correct.